src/esw/brushed_motors/Controller.cpp

```diff
@@ -76,12 +94,85 @@ void Controller::moveOpenLoop(float input) {
 
         I2C::transact(deviceAddress, motorIDRegMask | OPEN_PLUS_OP, OPEN_PLUS_WB,
                       OPEN_PLUS_RB, buffer, UINT8_POINTER_T(&angle));
-        currentAngle = (float) ((((float) angle / quadCPR) * 2 * M_PI) - M_PI);
+        currentAngle = (float) (((float) angle / quadCPR) * 2 * M_PI);
     } catch (IOFailure& e) {
         ROS_ERROR("moveOpenLoop failed on %s", name.c_str());
     }
 }
 
+// REQUIRES: nothing
+// MODIFIES: nothing
+// EFFECTS: returns if the MCU is calibrated
+bool Controller::isCalibrated() {
+    uint8_t calibration_status;
+
+    try {
+        makeLive();
+
+        I2C::transact(deviceAddress, motorIDRegMask | IS_CALIBRATED_OP, IS_CALIBRATED_WB,
+                      IS_CALIBRATED_RB, nullptr, UINT8_POINTER_T(&calibration_status));
+
+    } catch (IOFailure& e) {
+        ROS_ERROR("isCalibrated failed on %s", name.c_str());
+    }
+
+    return calibration_status;
+}
+
+// REQUIRES: nothing
+// MODIFIES: nothing
+// EFFECTS: enables or disables limit switches
+void Controller::enableLimitSwitches(bool enable) {
+    try {
+        makeLive();
+
+        bool enableLimitA = limitAPresent && enable;
+
+        uint8_t buffer[1];
+        memcpy(buffer, UINT8_POINTER_T(&enableLimitA), sizeof(enableLimitA));
+        I2C::transact(deviceAddress, motorIDRegMask | ENABLE_LIMIT_A_OP, ENABLE_LIMIT_A_WB,
+                      ENABLE_LIMIT_A_RB, buffer, nullptr);
+
+        bool enableLimitB = limitBPresent && enable;
+
+        memcpy(buffer, UINT8_POINTER_T(&enableLimitB), sizeof(enableLimitB));
+        I2C::transact(deviceAddress, motorIDRegMask | ENABLE_LIMIT_B_OP, ENABLE_LIMIT_B_WB,
+                      ENABLE_LIMIT_B_RB, buffer, nullptr);
+        
+
+    } catch (IOFailure& e) {
+        ROS_ERROR("enableLimitSwitches failed on %s", name.c_str());
+    }
+}
+
+
+// REQUIRES: nothing
+// MODIFIES: nothing
+// EFFECTS: gets current absolute encoder value of MCU
+float Controller::getAbsoluteEncoderValue() {
+    try {
+        makeLive();
+
+        float abs_enc_radians;
+        I2C::transact(deviceAddress, motorIDRegMask | ABS_ENC_OP, ABS_ENC_WB,
+                      ABS_ENC_RB, nullptr, UINT8_POINTER_T(&abs_enc_radians));
+
+        return abs_enc_radians;
+
+    } catch (IOFailure& e) {
+        ROS_ERROR("getAbsoluteEncoderValue failed on %s", name.c_str());
+    }
+
+    return 0;
+}
+
+// REQUIRES: nothing
+// MODIFIES: nothing
+// EFFECTS: Returns true if Controller has a (one or both) limit switch(s) is enabled.
+bool Controller::getLimitSwitchEnabled() const {
+    return limitAPresent || limitBPresent;
```

Comment:
Just because two limits are present does not mean that is enabled. We need to maintain a variable that keeps track of if limits are enabled or not.